mesheryctl/internal/cli/root/filter/filter_test.go

```diff
@@ -0,0 +1,140 @@
+package filter
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+)
+
+func TestFilterCmd(t *testing.T) {
+	utils.SetupContextEnv(t)
+
+	// initialize mock server for handling requests
+	utils.StartMockery(t)
+
+	// create a test helper
+	testContext := utils.NewTestHelper(t)
+
+	// get current directory
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Fatal("Not able to get current working directory")
+	}
+	currentDir := filepath.Dir(filename)
+	fixturesDir := filepath.Join(currentDir, "fixtures")
+
+	FilterTestCases := []struct {
+		Name             string
+		Args             []string
+		URL              string
+		Fixture          string
+		Token            string
+		ExpectedResponse string
+		ExpectHelp       bool
+		ExpectErr        bool
+	}{
+		{
+			Name:             "filter viewcmd with name",
+			Args:             []string{"view", "view-filter-name"},
+			Token:            filepath.Join(fixturesDir, "token.golden"),
+			ExpectedResponse: "filter.id.view.output.golden",
+			Fixture:          "filter.name.view.api.response.golden",
+			URL:              testContext.BaseURL + "/api/filter",
+			ExpectHelp:       false,
+			ExpectErr:        false,
+		},
+		{
+			Name:             "filter viewcmd with ID",
+			Args:             []string{"view", "c0c6035a-b1b9-412d-aab2-4ed1f1d51f84"},
```

Comment:
can we have a test for invalid ID as well?

Reply:
Done